### PR TITLE
App component

### PR DESCRIPTION
**Pull request by fitzsimonsdevin**
# Summary of changes
Likely should have made this a touch smaller. Didn't really expect the tutorial to have me touching multiple components that quickly.
- Establish App test, styles and component
- Establish Calculator, styles test and component. 

# Acceptance Criteria
- [x] My code follows the style guidelines of this project
- [ ] I have run eslint and prettier and have no errors in my files
- [x] I have performed a self-review of my own code
- [ ] CodeClimate has no errors. If there are CodeClimate errors that you choose not to fix, please note why below and initiate a discussion with the team concerning the validity of the errors.

### CODE REVIEW

File: src/components/App/App.spec.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+describe('App', () => {
+    let wrapper;
+    beforeEach(() => {
+        wrapper = shallow(<App />);
+    })
+
+    it ('should render correctly', () => expect(wrapper).toMatchSnapshot())
+    
+    it ('should render a div', () => {
```
Comment (kyleruschel) [inline src/components/App/App.spec.js]:
```suggestion
    it ('renders an app-container div', () => {
```

File: src/components/App/App.spec.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import App from './App';
+import Calculator from '../Calculator/Calculator';
+
+describe('App', () => {
+    let wrapper;
+    beforeEach(() => {
+        wrapper = shallow(<App />);
+    })
+
+    it ('should render correctly', () => expect(wrapper).toMatchSnapshot())
```
Comment (kyleruschel) [inline src/components/App/App.spec.js]: Good news, we don't do snapshot testing here! Feel free to continue using if you're accustomed to doing so, but it's not required here at Nutrien.

File: src/components/App/App.spec.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    
+    it ('should render a div', () => {
+        expect(wrapper.find('div').length).toEqual(1);
```
Comment (kyleruschel) [inline src/components/App/App.spec.js]:
```suggestion
        expect(wrapper.find('div').hasClass('app-container')).toBe(true);
```

File: src/components/App/App.spec.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import App from './App';
+import Calculator from '../Calculator/Calculator';
```
Comment (kyleruschel) [inline src/components/App/App.spec.js]: We can mock our dependencies to increase our test performance. Here is a good example. We want to verify that the App component returns the `Calculator` component, but we don't need to import the component at all. We just need to mock it, and test for the mocks instance. We can do this like this: 

```suggestion
jest.mock('../Calculator/Calculator', () => 'Calculator');
```

File: src/components/App/App.spec.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    })
+
+    it ('should render correctly', () => expect(wrapper).toMatchSnapshot())
+    
+    it ('should render a div', () => {
+        expect(wrapper.find('div').length).toEqual(1);
+    });
+
+    it ('should contain a Calculator Element', () => {
+        expect(wrapper.containsMatchingElement(<Calculator />)).toEqual(true)
```
Comment (kyleruschel) [inline src/components/App/App.spec.js]: Now that we've mocked at the top, we can simply check for the string of Calculator.
```suggestion
        expect(wrapper.find('Calculator').length).toBe(1)
```

File: src/components/App/__snapshots__/App.spec.js.snap [hunk 6]
```diff
@@ -0,0 +1,83 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (kyleruschel) [inline src/components/App/__snapshots__/App.spec.js.snap]: 👍 up to you to leave or remove

File: src/components/Calculator/Calculator.spec.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import Calculator from './Calculator';
+
+describe('Calculator', () => {
+    let wrapper;
+    beforeEach(() => wrapper = shallow(<Calculator />))
```
Comment (kyleruschel) [inline src/components/Calculator/Calculator.spec.js]:
```suggestion
```

Comment (kyleruschel) [inline src/components/Calculator/Calculator.spec.js]: I can see you're truly following TDD here, but we can do a little refactor now. The tests on line 8 and 12 are the same, and therefore we can eliminate the first. By doing so, we can refactor this to remove the beforeEach, as it is no longer necessary.

File: src/components/Calculator/Calculator.spec.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+    it('should render a div', () => {
+        expect(wrapper.find('div').length).toEqual(1);
+    });
+
+    it('should have the class for Calculator', () => {
```
Comment (kyleruschel) [inline src/components/Calculator/Calculator.spec.js]: On EXH, we've chosen to remove `should` from our it blocks to show confidence in our tests. It also reads better grammatically once you begin to become more descriptive in your conditionals

```suggestion
    it('contains a div with className "calculator-container"', () => {
```

File: src/components/App/App.spec.js (hunk 1 above)
Comment (fitzsimonsdevin) [reply src/components/App/App.spec.js]: I'll address this, and a few others in a single PR that aligns my project more closely with EXH best practices

**Pull request merged**